### ISSUE TITLE
Multiple rules have false positives inside blockquotes

### ISSUE THREAD

**Issue by joshrotenberg** [joshrotenberg/mdbook-lint#275]
## Summary

Several rules incorrectly flag content inside blockquotes (`>`) because they don't properly handle blockquote context.

### MD022 (blanks-around-headings)
Flags headings inside blockquotes even when proper blank lines exist:
```markdown
> ### Command Line Notation
>
> In this chapter...
```
We flag line 17 but there IS a blank line (the `>` only line).

### MD031 (blanks-around-fences)
Flags code fences inside blockquotes:
```markdown
> <Listing file-name="src/main.rs">
>
> '''rust,ignore
> struct User {
```
We flag the code fence but it's properly surrounded by blank lines within the blockquote.

## Impact

| Rule | mdbook-lint | rumdl | markdownlint |
|------|-------------|-------|--------------|
| MD022 | 14 | 0 | 0 |
| MD031 | 12 | 0 | 0 |

## Root Cause

These rules check for blank lines but don't account for blockquote markers. A line with just `>` IS a blank line within a blockquote.

## Related

Issue #263 - Linter comparison
